apps/trading/components/swap/swap-container.tsx

```diff
@@ -14,7 +14,13 @@ import { useDataProvider } from '@vegaprotocol/data-provider';
 import { SwapForm } from './swap-form';
 import { Loader, Splash } from '@vegaprotocol/ui-toolkit';
 
-export const SwapContainer = ({ assetId }: { assetId?: string }) => {
+export const SwapContainer = ({
+  assetId,
+  onDeposit,
+}: {
+  assetId?: string;
+  onDeposit: (assetId?: string) => void;
```

Comment:
This component is within apps/trading, so I would just avoid prop drilling and use useNavigate directly in the component with the click handler

Reply:
super agree! removed